owncloudComLibrary/src/test/java/com/owncloud/android/lib/StatusRequestorTest.kt

```diff
@@ -0,0 +1,41 @@
+package com.owncloud.android.lib
+
+import com.owncloud.android.lib.resources.status.StatusRequestor
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class StatusRequestorTest {
+    private val requestor = StatusRequestor()
+
+    @Test
+    fun testUpdateLocationWithAbsolutePath() {
+        val newLocation = requestor.updateLocationWithRedirectPath(
+            "https://cloud.somewhere.com", "https://cloud.somewhere.com/subdir"
+        )
+        assertEquals("https://cloud.somewhere.com/subdir", newLocation)
+    }
+
+    @Test
+    fun testUpdateLocationWithAbsolutePathBackwards() {
+        val newLocation = requestor.updateLocationWithRedirectPath(
+            "https://cloud.somewhere.com/subdir", "https://cloud.somewhere.com/"
+        )
+        assertEquals("https://cloud.somewhere.com/", newLocation)
+    }
+
+    @Test
+    fun testUpdateLocationWithRelativePath() {
+        val newLocation = requestor.updateLocationWithRedirectPath(
+            "https://cloud.somewhere.com", "/subdir"
+        )
+        assertEquals("https://cloud.somewhere.com/subdir", newLocation)
+    }
+
+    @Test
+    fun testUpdateLocationWithRelativePathSidewards() {
+        val newLocation = requestor.updateLocationWithRedirectPath(
+            "https://cloud.somewhere.com/some/other/subdir", "/subdir"
+        )
+        assertEquals("https://cloud.somewhere.com/subdir", newLocation)
```

Comment:
`"https://cloud.somewhere.com` comes with a taste of a variable to use it everywhere

Reply:
I agree!

Reply:
done